app/src/main/java/com/example/githubusers/data/repository/Source.kt

```diff
@@ -0,0 +1,58 @@
+package com.example.githubusers.data.repository
+
+import androidx.paging.PagingSource
+import androidx.paging.PagingState
+import com.example.githubusers.data.api.ApiService
+import com.example.githubusers.data.logicData.LogicDataRoom
+import com.example.githubusers.model.UsersItemEntity
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import timber.log.Timber
+import javax.inject.Inject
+
+class Source @Inject constructor(
+    private val logicDataRoom : LogicDataRoom,
+    private val api : ApiService
+): PagingSource<Int, UsersItemEntity>() {
+
+    override fun getRefreshKey(state: PagingState<Int, UsersItemEntity>): Int {
+
+        Timber.d("state.pages.last().data.last().id: ${state.pages.last().data.last().id}")
+
+        return state.pages.last().data.last().id
```

Comment:
probably the better implementation of the method is here `https://medium.com/androiddevelopers/introduction-to-paging-3-0-in-the-mad-skills-series-648f77231121`